packages/account/tests/SmartAccountV2.local.spec.ts

```diff
@@ -361,8 +367,243 @@ describe("BiconomySmartAccountV2 API Specs", () => {
+  //   })
+
+  //   const module = await ECDSAOwnershipValidationModule.create({
+  //     signer: owner,
+  //     moduleAddress: DEFAULT_ECDSA_OWNERSHIP_MODULE
+  //   })
+
+  //   const newAccount = await BiconomySmartAccountV2.create({
+  //       chainId: ChainId.POLYGON_MUMBAI,
+  //       entryPointAddress: DEFAULT_ENTRYPOINT_ADDRESS,
+  //       signer: owner,
+  //       bundler,
+  //       defaultValidationModule: module,
+  //       activeValidationModule: module
+  //   });
+
+  //   const accountAddress: `0x${string}` = await newAccount.getAccountAddress() as `0x${string}`;
+
+  //   const prefundResp = await walletClient.sendTransaction({
+  //     account: wallet,
+  //     to:  accountAddress,
+  //     value: 100000000000000000n,
+  //     chain: polygonMumbai
+  //   });
+    
+  //   const tx = {
+  //     to: await Wallet.createRandom().getAddress(),
+  //     data: "0x"
+  //   }
+
+  //   const userOp = await newAccount.buildUserOp([tx]);
+  //   const res = await newAccount.sendUserOp(userOp);
+  //   const txhash = await res.waitForTxHash();
+
+  //   console.log("txhash ", txhash);
+
+  //   expect(txhash).toBeDefined();
+  // });
+
+  it("Create smart account with default module (ECDSA) without creating instance or providing module name", async () => {
+
+    const account: BiconomySmartAccountV2 = await BiconomySmartAccountV2.create({
+      chainId: ChainId.GANACHE,
+      rpcUrl: "http://127.0.0.1:8545",
+      entryPointAddress: entryPoint.address,
+      signer,
+    });
+
+    const address = await account.getAccountAddress();
+    console.log("Module Abstraction Test - Account address ", address);
+
+    expect(address).toBe(account.accountAddress);
+
+    const module = account.activeValidationModule;
+    console.log(`ACTIVE MODULE - ${module.getAddress()}`);
+
+    expect(module.getAddress()).toBe(DEFAULT_ECDSA_OWNERSHIP_MODULE);
+
+  }, 10000);
+
+  it("Create smart account with ECDSA module without creating instance", async () => {
+
+    const account: BiconomySmartAccountV2 = await BiconomySmartAccountV2.create({
+      chainId: ChainId.GANACHE,
+      rpcUrl: "http://127.0.0.1:8545",
+      entryPointAddress: entryPoint.address,
+      signer,
+    });
+
+    const address = await account.getAccountAddress();
+    console.log("Module Abstraction Test - Account address ", address);
+
+    expect(address).toBe(account.accountAddress);
+
+    const module = account.activeValidationModule as ECDSAOwnershipValidationModule;
+    
+    console.log(`ACTIVE MODULE - ${module.getAddress()}`);
+
+    expect(module.getAddress()).toBe(DEFAULT_ECDSA_OWNERSHIP_MODULE);
+  }, 10000);
+
+  it("Create smart account with default module using WalletClientSigner as signer", async () => {
+
+    const walletClient = createWalletClient({
+      chain: localhost, 
+      transport: http("http://127.0.0.1:8545"),
+    });
+    
+    const smartAccountSigner: SmartAccountSigner = new WalletClientSigner(
```

Comment:
aren't these two different types?

Reply:
Does casting matter so much in a test file?

Reply:
no, it was just confusing to me. 
like what is the need to import SmartaccountSigner from alchemy?
I haven't seen the implementation but what is the difference between that and WalletClientSigner and requirement for casting here.

because later you're passing smartAccountSigner to account config
`signer: smartAccountSigner,`

and for this as per current defined types it's Signer | WalletClientSigner

Reply:
Oh I see, gotcha. Yeah it looks like it doesn't need to be cast here